### DISCUSSION TITLE
String Input sets itself to "0" on empty string

### DISCUSSION THREAD

**Discussion by jnslxndr** [cables-gl/cables_docs#768] [! Welcome - readme first!]
Hi there

I noticed that (unconnected) String inputs on Ops get set to a string "0" when I set it to an empty string. Is this expected behaviour?

When removing tokens from a string using StringReplace, how to I deal with this behaviour? Do I connect an extra String Op to enforce an empty string?

Cheers ~ jens

![image](https://github.com/cables-gl/cables_docs/assets/9602/a0357a69-3efb-4c89-a97b-b30cca98179c)

**Comment by pandrr** [comment]
how/when does this happen ? 

when i delete the default content the input field is empty, it still is after saving+reloading etc.

**Reply by jnslxndr** [reply]
Thanks for looking into it. I haven't had time since the last project to make a patch which reproduces the behaviour. However, I can confirm this happened a number times in different patches. I usually connected a String Op to enforce an empty string.

Today I tried to make a new and isolated patch to reproduce the bug, but not successfully so far. Was there an update which might have solved the issue?

I'll close for now and come back when it reappears in the next project.

**Comment by jnslxndr** [comment]
Hi @pandrr 

Sorry to reopen this Heisenbug. It occurred again multiple times - on other Ops than String Replace. It is very hard to reproduce, though. One thing i noticed is that it was tuck in that behaviour once the "0" was in the input port. Not trusting my memory 100%, but it could be that disconnecting a cable is the initial trigger.

While in some cases it can be mitigated or is irrelevant, it still is annoying behaviour to navigate arround.
